### ISSUE TITLE
High CPU load with fluidPage and conditionalPanel after upgrade to shiny 0.11

### ISSUE THREAD

**Issue by vzemlys** [rstudio/shiny#717]
I've noticed that my shiny apps run slower after upgrade to shiny 0.11. I've managed to create a reproducible example. 

The app with low CPU load:
```
library(shiny)
library(ggvis)
library(dplyr)

ui <- shinyUI(bootstrapPage(
    		     ggvisOutput("devOrA"),
			      uiOutput("devOrA_ui"),
			      ggvisOutput("devOrB"),
			      uiOutput("devOrB_ui"),
  ggvisOutput("devOrG"),
  uiOutput("devOrG_ui")))
   
server <- function(input, output, session) {

  # to deal correctly with milliseconds in timestamp
  options(digits=15)

  connectToStream <- reactive({

    invalidateLater(3000, session)

    data <- data.frame(sensorname = "screenAdjustedEvent",
                       x=1:10,y=runif(10, 5.0, 7.5),z=1:10,
                       recordTime=as.numeric(Sys.time())-(1:10),
                       device="iPhone", cognitoId="abc",
                       ip="dddd")
    data
  })

  # Device Orientation
  devOrA <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrA %>%
    ggvis(x = ~recordTime, y = ~x) %>%
    bind_shiny("devOrA","devOrA_ui", session)

  devOrB <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrB %>%
    ggvis(x = ~recordTime, y = ~x) %>%
    bind_shiny("devOrB", "devOrB_ui", session)

  devOrG <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrG %>%
    ggvis(x = ~recordTime, y = ~z) %>%
    bind_shiny("devOrG","devOrG_ui", session)
}

shinyApp(ui, server)

```

The app with high CPU load:
```
library(shiny)
library(ggvis)
library(dplyr)

ui <- shinyUI(
    fluidPage(
        titlePanel(
            h4("Test CPU load"), "Test CPU load"
            ),
        sidebarLayout(
            sidebarPanel(                
                conditionalPanel(
                    condition="input.condPanels==1",
                    h5("Select vertical axis"),                  
                    h5("Select horizontal axis"),
                    h5("Conversions"),
                    checkboxInput("c_yaxis",label="Same conversions for both axes",value=TRUE),
		                     radioButtons("cx_trans",label="",choices=c("Actuals"=1,"Log"=4),selected=1,inline=TRUE),
		    radioButtons("cx_norm",label="",choices=c("Total"=1,"Per capita"=2,"Per household"=3,"Share of"=4),selected=1,inline=TRUE)
		                 
                   ),
                conditionalPanel(condition="input.condPanels==4",
                                 downloadButton('c_download', 'Download data')
                                 )
                ),            
                mainPanel(
                tabsetPanel(                  
                    tabPanel("Compare countries",
			     ggvisOutput("devOrA"),
			      uiOutput("devOrA_ui"),
			      ggvisOutput("devOrB"),
			      uiOutput("devOrB_ui"),
  ggvisOutput("devOrG"),
  uiOutput("devOrG_ui"),
                            value=1),
                    tabPanel("Data",                           
                             tableOutput("Data"),
                             value=4
                             ),
                    id="condPanels"
                    )
                )
            )
        )
    )	



server <- function(input, output, session) {

  # to deal correctly with milliseconds in timestamp
  options(digits=15)

  connectToStream <- reactive({

    invalidateLater(3000, session)

    data <- data.frame(sensorname = "screenAdjustedEvent",
                       x=1:10,y=runif(10, 5.0, 7.5),z=1:10,
                       recordTime=as.numeric(Sys.time())-(1:10),
                       device="iPhone", cognitoId="abc",
                       ip="dddd")
    data
  })

  # Device Orientation
  devOrA <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrA %>%
    ggvis(x = ~recordTime, y = ~x) %>%
    bind_shiny("devOrA","devOrA_ui", session)

  devOrB <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrB %>%
    ggvis(x = ~recordTime, y = ~x) %>%
    bind_shiny("devOrB", "devOrB_ui", session)

  devOrG <- reactive({
    streamdata <- connectToStream()

    streamdata %>%
      filter(sensorname == "screenAdjustedEvent") %>%
      filter(recordTime > max(recordTime) - 30)
  })
  devOrG %>%
    ggvis(x = ~recordTime, y = ~z) %>%
    bind_shiny("devOrG","devOrG_ui", session)
}

shinyApp(ui, server)


```

Note that I've only changed the UI, by adding some checkboxes and radioButtons.

I've tested on R 3.1.1 on Mac OS X 10.9.5 on Safari. I've used an eye test of CPU usage, for the first app, the CPU load was practically non-existent, for the second app the CPU load rised up to 20%. 

For the app I develop, the CPU usage is much higher. On Firefox it reached about 90%, about 40% for Safari.

**Comment by jcheng5** [comment]
Just to be clear, the CPU load increase is for the R process, or the browsers?

**Comment by vzemlys** [comment]
The browser process. R load is practically non-existent.

**Comment by wch** [comment]
This might be related to #693.

**Comment by erichare** [comment]
We are seeing this too - We're not sure if it's related to navbarPage or not, but we are using a navbarPage in our app intRo and seeing the same slowdown. See https://erichare.shinyapps.io/intRo (Shiny 0.10.2.2) vs. https://erichare.shinyapps.io/intRo-dev (Shiny 0.11). Clicking the graphical tab and then switching X Variable for the histogram displays the slowdown pretty clearly. On my machine, the 0.11 version of our app uses about 40% CPU (browser process, not R process) for the first 2 minutes or so on load, while the older version uses nearly 0. 

The code between the two apps is identical except that we modified some of the custom CSS to be bootstrap3 compliant.

**Comment by jcheng5** [comment]
This will be fixed today, sorry for the inconvenience!

**Comment by mostly-harmless** [comment]
Just installed the testfix branch and am seeing the same improvements in speed that Eric mentioned. It really may be faster than .10.2.2. Mathjax rendering seems a lot faster as well




Thank you!!!


—

**Comment by vzemlys** [comment]
Yes I can confirm that with a fix, high cpu load has disappeared. Thanks!

**Comment by wch** [comment]
We merged a fix for this into master (#723). If you could reinstall and test it out, we'd appreciate it! Please let us know about performance and if anything odd happens.

```R
devtools::install_github('rstudio/shiny')
```

**Comment by vzemlys** [comment]
Everything works as expected, for now :) Thanks!
